### PR TITLE
Set up CI for Python package with GitHub Actions

### PR DESCRIPTION
**Pull request by PokhodenkoSA**
This PR creates CI based on GitHub Actions.
- Based on scripts from https://github.com/oneapi-src/oneapi-ci
- Use oneAPI 2021.2
- Only Linux yet
- Build with `python setup.py develop`
- Use Python 3.7, 3.8, 3.9
- Use cache in order to install onaAPI in a few seconds
- Run pytest
- With TBB activated. CPU device is available.
- GPU device is not available.
- Closes #359 because contains it. #359 fixes tests which should be skipped if GPU is not available.
- I would prefer squash this PR.

### CODE REVIEW

File: scripts/build_backend.py [hunk 1]
```diff
@@ -85,7 +85,7 @@
             "-DDPCPP_INSTALL_DIR=" + DPCPP_ROOT,
             "-DCMAKE_C_COMPILER:PATH=" + os.path.join(DPCPP_ROOT, "bin", "clang"),
             "-DCMAKE_CXX_COMPILER:PATH=" + os.path.join(DPCPP_ROOT, "bin", "dpcpp"),
-            "-DDPCTL_ENABLE_LO_PROGRAM_CREATION=ON",
+            # "-DDPCTL_ENABLE_LO_PROGRAM_CREATION=ON",
```
Comment (PokhodenkoSA) [inline scripts/build_backend.py:88]: It is blocker for the PR and should be fixed before merging this PR.
LevelZero is not available.

File: scripts/cache_exclude_linux.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
```
Comment (PokhodenkoSA) [inline scripts/cache_exclude_linux.sh:1]: From https://github.com/oneapi-src/oneapi-ci

File: scripts/install_linux.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
```
Comment (PokhodenkoSA) [inline scripts/install_linux.sh:1]: This script from https://github.com/oneapi-src/oneapi-ci

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+Cython
+numpy
+pytest
+pytest-cov
```
Comment (PokhodenkoSA) [inline requirements.txt:4]: Created `requirements.txt` for setup python environment.

File: .github/workflows/python-package.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+        path: |
+          /opt/intel/oneapi/compiler
+          /opt/intel/oneapi/tbb
+        key: build-${{ env.LINUX_BASEKIT_URL }}-${{ env.LINUX_DPCPP_COMPONENTS_WEB }}-compiler-tbb-${{ hashFiles('**/scripts/cache_exclude_linux.sh') }}
+    - name: Install oneAPI
+      if: steps.cache-oneapi.outputs.cache-hit != 'true'
+      run: scripts/install_linux.sh $LINUX_BASEKIT_URL $LINUX_DPCPP_COMPONENTS_WEB
+    - name: Exclude unused files from oneAPI cache
+      if: steps.cache-oneapi.outputs.cache-hit != 'true'
+      run: scripts/cache_exclude_linux.sh
```
Comment (PokhodenkoSA) [inline .github/workflows/python-package.yml:46]: This make oneAPI folder from 800Mb to 330Mb.

File: .github/workflows/python-package.yml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
+
+    env:
+      ONEAPI_ROOT: /opt/intel/oneapi
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
```
Comment (PokhodenkoSA) [inline .github/workflows/python-package.yml:22]: It fetches whole repository history. Is not necessary but with it we can see package version based on tag.

File: scripts/build_backend.py (hunk 1 above)
Comment (PokhodenkoSA) [reply scripts/build_backend.py:88]: Ticket for fixing it https://github.com/IntelPython/dpctl/issues/361

**Comment by PokhodenkoSA** [comment]
Convert to draft because problem with LevelZero (#361) is not fixed yet.

**Comment by diptorupd** [comment]
😃👍🏼👍🏼

File: .github/workflows/python-package.yml [hunk 7]
```diff
@@ -0,0 +1,59 @@
+name: Python package
+
+on:
+  pull_request:
+  push:
+    branches: [master]
+
+env:
+  LINUX_BASEKIT_URL: https://registrationcenter-download.intel.com/akdlm/irc_nas/17769/l_BaseKit_p_2021.2.0.2883_offline.sh
```
Comment (diptorupd) [inline .github/workflows/python-package.yml:9]: Why use this hard coded url? Instead we should just use the [APT oneAPI repo](https://software.intel.com/content/www/us/en/develop/documentation/installation-guide-for-intel-oneapi-toolkits-linux/top/installation/install-using-package-managers/apt.html?fireglass_rsn=true). Github actions allow using password-less `sudo`.

Another option can be to use nightly builds of dpcpp from https://github.com/intel/llvm/releases. I am working on a set of clean ups to our build system that will make it very easy to build with the open source dpcpp.

**Comment by PokhodenkoSA** [comment]
Deprecated. See #515.

**Pull request closed without merging**